crates/ruma-common/src/events/room/message/sanitize.rs

```diff
@@ -0,0 +1,293 @@
+];
+
+/// The HTML tag name for a rich reply fallback.
+pub const RICH_REPLY_TAG: &str = "mx-reply";
+
+/// Allowed attributes per HTML tag according to the Matrix specification.
+pub const ALLOWED_ATTRIBUTES_STRICT: [(&str, &[&str]); 6] = [
+    ("font", &["data-mx-bg-color", "data-mx-color", "color"]),
+    ("span", &["data-mx-bg-color", "data-mx-color", "data-mx-spoiler"]),
+    ("a", &["name", "target", "href"]),
+    ("img", &["width", "height", "alt", "title", "src"]),
+    ("ol", &["start"]),
+    ("code", &["class"]),
+];
+
+/// Allowed schemes of URIs per HTML tag and attribute tuple according to the Matrix specification.
+pub const ALLOWED_SCHEMES_STRICT: [((&str, &str), &[&str]); 2] =
+    [(("a", "href"), &["http", "https", "ftp", "mailto", "magnet"]), (("img", "src"), &["mxc"])];
+
+/// Extra allowed schemes of URIs per HTML tag and attribute tuple.
+///
+/// This is a convenience list to add schemes that can be encountered but are not listed in the
+/// Matrix specification. It consists of:
+///
+/// * The `matrix` scheme for `a` tags (see [matrix-org/matrix-spec#1108]).
+///
+/// To get a complete list, add these to `ALLOWED_SCHEMES_STRICT`.
+///
+/// [matrix-org/matrix-spec#1108]: https://github.com/matrix-org/matrix-spec/issues/1108
+pub const ALLOWED_SCHEMES_COMPAT: [((&str, &str), &[&str]); 1] = [(("a", "href"), &["matrix"])];
+
+/// Allowed classes per HTML tag according to the Matrix specification.
+pub const ALLOWED_CLASSES_STRICT: [(&str, &[&str]); 1] = [("code", &["language-*"])];
+
+/// Max depth of nested HTML tags allowed by the Matrix specification.
+pub const MAX_DEPTH_STRICT: u32 = 100;
+
+/// Sanitize the given HTML string.
+///
+/// This removes the [tags and attributes] that are not listed in the Matrix specification.
+///
+/// It can also optionally remove the [rich reply fallback].
+///
+/// [tags and attributes]: https://spec.matrix.org/v1.2/client-server-api/#mroommessage-msgtypes
+/// [rich reply fallback]: https://spec.matrix.org/v1.2/client-server-api/#fallbacks-for-rich-replies
+#[cfg(feature = "sanitize")]
+pub fn sanitize_html(
+    s: &str,
+    mode: HtmlSanitizerMode,
+    remove_reply_fallback: RemoveReplyFallback,
+) -> String {
+    let sanitizer = if mode == HtmlSanitizerMode::Compat {
+        HtmlSanitizer::compat(remove_reply_fallback)
+    } else {
+        HtmlSanitizer::new(remove_reply_fallback)
+    };
+    sanitizer.clean(s)
+}
+
+/// What HTML [tags and attributes] should be kept by the sanitizer.
+///
+/// [tags and attributes]: https://spec.matrix.org/v1.2/client-server-api/#mroommessage-msgtypes
+#[cfg(feature = "sanitize")]
+#[derive(Clone, Copy, Debug, PartialEq, Eq)]
+#[allow(clippy::exhaustive_enums)]
+pub enum HtmlSanitizerMode {
+    /// Keep only the tags and attributes listed in the Matrix specification.
+    ///
+    /// This uses the `STRICT` constants.
+    Strict,
+
+    /// Keeps all the tags ant attributes in `Strict` mode, and others that are not in this section
```

Comment:
No ants in our docs please 😉 

```suggestion
    /// Keeps all the tags and attributes in `Strict` mode, and others that are not in this section
```